test/integration/JsonParsing.test.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  PrimitiveUri,
+  IElement,
+  IExtension,
+  Extension,
+  IKitchenSink,
+  KitchenSink,
+  PrimitiveTime,
+  Coding,
+  PrimitiveBoolean,
+  IResourceChild,
+  Resource,
+  ResourceChild, Bundle, BundleEntry,
+} from "../fixture/generatedTypeScript/fhir";
+import { IBundle } from "../fixture/generatedTypeScript/interfaces/IBundle";
+
+describe("Testing the Parsing logic of a generated class", () => {
+  function assertExtension(extensions?: Array<Extension>): void {
+    expect(extensions).toBeArrayOfSize(1);
+    const [extension] = extensions as Array<Extension>;
+    expect(PrimitiveBoolean.isPrimitiveBoolean(extension.value)).toBeTrue();
+    if (PrimitiveBoolean.isPrimitiveBoolean(extension.value)) {
+      expect(extension.value.value).toBeTrue();
+    }
+  }
+
+  it("should work around any circular dependencies", () => {
+    const element: IElement = { id: "val" };
+    expect(element.id).toBe("val");
+
+    const extension: IExtension = { valueString: "valString" };
+    expect(extension.valueString).toBe("valString");
+  });
+
+  it("should parse primitive JSON", () => {
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5923420)